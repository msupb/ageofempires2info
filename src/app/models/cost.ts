export interface ICost {
    info?: string;
    Wood?: number;
    Food?: number;
    Stone?: number;
    Gold?: number;
}