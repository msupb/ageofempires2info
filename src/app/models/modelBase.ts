export interface IModelBase {
    localId?: string;
    category?: string;
    id: number;
    name: string;
}