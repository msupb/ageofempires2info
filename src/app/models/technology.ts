import { ICost } from "./cost";

export interface ITechnology {
    id: number;
    name: string;
    description: string;
    expansion: string;
    age: string;
    develops_in: string;
    cost: ICost;
    build_time: number;
    applies_to: Array<string>;
}