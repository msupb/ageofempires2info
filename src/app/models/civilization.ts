import {IModelBase} from './modelBase';

export interface ICivilization extends IModelBase {
    expansion: string;
    army_type: string;
    unique_unit: Array<string>;
    unique_tech: Array<string>;
    team_bonus: string;
    civilization_bonus: Array<string>;
}