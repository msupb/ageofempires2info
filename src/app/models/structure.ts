import { ICost } from "./cost";
import {IModelBase} from './modelBase';

export interface IStructure extends IModelBase {
    description: string;
    expansion: string;
    age: string;
    cost: ICost;
    build_time: number;
    hit_points: number;
    line_of_sight: number;
    armor: string;
    range: string;
    relaod_time: number;
    attack: number;
    special: Array<string>
}