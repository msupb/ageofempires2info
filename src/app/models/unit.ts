import { ICost } from "./cost";
import {IModelBase} from './modelBase';

export interface IUnit extends IModelBase {
    description: string;
    expansion: string;
    age: string;
    created_in: string;
    cost: ICost;
    build_time: number;
    reload_time: number;
    attack_delay: number;
    movement_rate: number;
    line_of_sight: number;
    hit_points: number;
    range: string;
    attack: number;
    armor: string;
    attack_bonus: Array<string>;
    armor_bonus: Array<string>;
    search_radius: number;
    accuracy: string;
    blast_radius: number;
}