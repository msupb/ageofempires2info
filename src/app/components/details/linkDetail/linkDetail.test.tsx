import React from 'react';
import { cleanup, render, within } from '@testing-library/react';
import LinkDetail from './linkDetail';
import { IModelBase } from '../../../models/modelBase';
import { BrowserRouter as Router } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import LinkDetailsComponent from './linkDetailsComponent';

const testList: Array<IModelBase> = [
    {
    'localId': 'testid',
    'category': 'testCategory',
    'id': 0,
    'name': 'testName',
    'age': 'testAge'
    }
]; 

// Test to see if LinkDetail contains a CusButton
afterEach(cleanup);

test("LinkDetailsComponent contains one <LinkDetail></LinkDetail>", () => {
    const { getAllByTestId, getByTestId } = render(<Router><LinkDetailsComponent links={['https://test.test.com/category/id']}></LinkDetailsComponent></Router>);
    const linkDetailsComponent = getByTestId('linkDetailsComponent');
    const linkDetailsInLinkDetailsComponent = within(linkDetailsComponent).getAllByTestId('linkDetail');
    expect(linkDetailsInLinkDetailsComponent.length).toBe(1);
})

it("matches LinkDetailsComponent snapshot", () => {
    const jsonObj = TestRenderer.create(<Router><LinkDetailsComponent links={['https://test.test.com/category/id']}></LinkDetailsComponent></Router>).toJSON();
    expect(jsonObj).toMatchSnapshot();
})

test("LinkDetail contains one <CusButton></CusButton>", () => {
    const { getAllByTestId, getByTestId } = render(<Router><LinkDetail itemList={testList}></LinkDetail></Router>);
    const linkDetail = getByTestId('linkDetail');
    const cusButtonInLinkDetail = within(linkDetail).getAllByTestId('cusButtonTest');
    expect(cusButtonInLinkDetail.length).toBe(1);
})

it("matches LinkDetail snapshot", () => {
    const jsonObj = TestRenderer.create(<Router><LinkDetail itemList={testList}></LinkDetail></Router>).toJSON();
    expect(jsonObj).toMatchSnapshot();
})




