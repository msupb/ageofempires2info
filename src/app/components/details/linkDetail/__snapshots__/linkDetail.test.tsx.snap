// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches LinkDetail snapshot 1`] = `
<div
  data-testid="linkDetail"
>
  <div
    className="container"
  >
    <h6>
      <b>
        testName
      </b>
    </h6>
    <div
      className="details-wrap"
    >
      <p
        className="link-details"
      >
        Age: 
        testAge
      </p>
      <a
        href="/details/testCategory/0"
        onClick={[Function]}
      >
        <button
          data-testid="cusButtonTest"
          onClick={[Function]}
          type="button"
        >
          Go To
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`matches LinkDetailsComponent snapshot 1`] = `
<div
  className="link-card card"
  data-testid="linkDetailsComponent"
/>
`;
